cysync/src/store/provider/transactionStatusProvider/txnStatusExecutor.ts

```diff
@@ -0,0 +1,259 @@
+import {
+  BtcCoinData,
+  COINS,
+  EthCoinData,
+  NearCoinData,
+  SolanaCoinData
+} from '@cypherock/communication';
+import { Status } from '@cypherock/database';
+import {
+  bitcoin as bitcoinServer,
+  eth as ethServer,
+  IRequestMetadata,
+  near as nearServer,
+  serverBatch as batchServer,
+  solana as solanaServer
+} from '@cypherock/server-wrapper';
+import { flatMap } from 'lodash';
+
+import logger from '../../../utils/logger';
+import { transactionDb } from '../../database';
+
+import { TxnStatusItem } from './txnStatusItem';
+
+export interface RequestMetaProcessInfo {
+  meta: IRequestMetadata[];
+  item: TxnStatusItem;
+  error?: Error;
+  isFailed: boolean;
+}
+
+export interface ExecutionResult {
+  item: TxnStatusItem;
+  isFailed: boolean;
+  isComplete?: boolean;
+  canRetry?: boolean;
+  error?: Error;
+  processResult?: any;
+  delay?: number;
+}
+
+export const getRequestsMetadata = (
+  item: TxnStatusItem
+): IRequestMetadata[] => {
+  const coin = COINS[item.coinType];
+
+  if (!coin) {
+    logger.warn('Invalid coin in sync queue', {
+      coinType: item.coinType
+    });
+    return undefined;
+  }
+
+  if (coin instanceof BtcCoinData) {
+    const metadata = bitcoinServer.transaction
+      .transactionStatus({ coinType: coin.abbr, hash: item.txnHash })
+      .getMetadata();
+    return [metadata];
+  }
+
+  if (coin instanceof EthCoinData) {
+    const metadata = ethServer.transaction
+      .transactionStatus(
+        { network: coin.network, txHash: item.txnHash },
+        item.isRefresh
+      )
+      .getMetadata();
```

Comment:
Can we create a separate file for the individual item?

Reply:
which items?